lib/ret/storage.ex

```diff
@@ -111,6 +111,18 @@ defmodule Ret.Storage do
     |> Enum.into(%{})
   end
 
+  # Similar to promote above, but allows for passing nil. Useful for optional upload fields
+  def promote_optional(map, %Account{} = account) when is_map(map) do
```

Comment:
I don't love this, but it matches a common usage pattern for us, and is pretty explicit... So I think I am ok with it.

Reply:
The code itself looks fine. The only issue I'd raise is that if there's a client bug in the future, or we open up these APIs for general use, it might be surprising if the server essentially fails silently if the client happens to pass nils when it didn't mean to. I think previously we would actually return a 404 via the `promotion_error` step in `scene_controller.ex`, so you've essentially changed the behavior of an existing API.
Not a strong opinion on this, just wanted to point it out.

Reply:
Yep thats right, though the DB constraints will still take care of ensuring that the required fields are always set at some point. As in, I wont be able to create a scene without a glb file for example. What does become allowed is updating a scene's thumbnail without also updating the glb for example... But that seems like a good thing. I do agree in general that this feels a bit weird, but I do think its expressing a usage pattern we seem to have in a few spots, so I think I am ok with it.